src/libraries/System.Text.RegularExpressions/tests/Regex.Match.Tests.cs

```diff
@@ -1092,10 +1093,15 @@ public async Task Match_SpecialUnicodeCharacters_Invariant(RegexEngine engine)
 
         [ConditionalTheory(nameof(IsNotArmProcessAndRemoteExecutorSupported))] // times out on ARM
         [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, ".NET Framework does not have fix for https://github.com/dotnet/runtime/issues/24749")]
-        [SkipOnCoreClr("Long running tests: https://github.com/dotnet/runtime/issues/10680", RuntimeConfiguration.Checked, RuntimeTestModes.JitMinOpts)]
+        [SkipOnCoreClr("Long running tests: https://github.com/dotnet/runtime/issues/10680", RuntimeConfiguration.Checked)]
         [MemberData(nameof(RegexHelpers.AvailableEngines_MemberData), MemberType = typeof(RegexHelpers))]
         public void Match_ExcessPrefix(RegexEngine engine)
         {
+            if (string.Equals(Environment.GetEnvironmentVariable("COMPlus_JITMinOpts"), "1", StringComparison.InvariantCulture))
```

Comment:
@AaronRobinsonMSFT, is it safe to use:
```suggestion
            if (string.Equals(Environment.GetEnvironmentVariable("DOTNET_JITMinOpts"), "1", StringComparison.InvariantCulture))
```
here? Locally it works.

Reply:
Is it worth moving this to a property in PlatformDetection?

Reply:
@am11 Yes, that will work for the product. However, superpmi that doesn't respect the `DOTNET_` prefix yet. I need to get to that at some point /cc @BruceForstall. The pattern should be clear with the new [`clrconfignocache.h`](https://github.com/dotnet/runtime/blob/main/src/coreclr/inc/clrconfignocache.h) but it needs to happen. See https://github.com/dotnet/runtime/blob/84a622590c9cfe37544cc3753029248f7e46067e/src/coreclr/ToolBox/superpmi/superpmi/jithost.cpp#L13-L46

Reply:
It makes sense.

After reading @stephentoub's comment, another option I was thinking of for a more permanent fix could be that we introduce something like `enum ConditionControl` with `LogicalOR`, `LogicalAnd` etc. in arcade, then change [this logic](https://github.com/dotnet/arcade/blob/3ea0d860c6973f2cbadc9e895c7ec2cbdaec4ad5/src/Microsoft.DotNet.XUnitExtensions/src/Discoverers/SkipOnCoreClrDiscoverer.cs#L29) to support both behaviors in the attribute. Other trait attributes (and their accompanying discovers in that directory) may also be able to take advantage of it. However, that kind of change would require to wait for the next Arcade update. :)

Reply:
Thanks @AaronRobinsonMSFT, I have updated it with `DOTNET_` prefix. Also, moved the condition to `PlatformDetection`.

Maybe we should also replace  `COMPlus_` with `DOTNET_` in SkipOnCoreClrDiscoverer (arcade repo).

Reply:
@am11 Well there are two issues here that we likely need to understand. The first is, do all of our build/ci scripts use `DOTNET_` as opposed to `COMPlus_`? I think the answer to that is "no", which means the above won't detect what it is trying to detect. The second is, if this is a feature that tests can branch on it might make more sense to bubble it up through the FeatureFlags system. I am dubious of that but it should be a conversation since the environment variable setting is so fluid.